hollow/src/main/java/com/netflix/hollow/core/index/HollowPrefixIndex.java

```diff
@@ -0,0 +1,438 @@
+
+            HollowObjectSchema objectSchema = (HollowObjectSchema) schema;
+            int position = objectSchema.getPosition(fields[i]);
+            if (position < 0)
+                throw new IllegalArgumentException("Invalid field position for field " + fields[i] + " in type " + tempType);
+
+            HollowObjectSchema.FieldType fieldType = objectSchema.getFieldType(fields[i]);
+            if (!(fieldType.equals(HollowObjectSchema.FieldType.REFERENCE) || fieldType.equals(HollowObjectSchema.FieldType.STRING)))
+                throw new IllegalArgumentException("Invalid field type for the field " + fields[i]);
+
+            fieldPositions[i] = position;
+            fieldTypes[i] = fieldType;
+            if (fieldType.equals(HollowObjectSchema.FieldType.REFERENCE)) {
+                tempType = objectSchema.getReferencedType(fields[i]);
+                if (typeSeen.contains(tempType))
+                    throw new IllegalStateException("Circular reference found in fieldPath for type " + tempType);
+            }
+            readState = (HollowObjectTypeReadState) readStateEngine.getTypeState(tempType);
+        }
+
+        // field path should ultimately lead down to a String type
+        if (!fieldTypes[fields.length - 1].equals(HollowObjectSchema.FieldType.STRING))
+            throw new IllegalArgumentException("Field path should resolve to a String type");
+
+        // get all cardinality to estimate size of array bits needed.
+        cardinalityOfKeyField = readState.getPopulatedOrdinals().cardinality();
+        HollowObjectTypeReadState valueState = (HollowObjectTypeReadState) readStateEngine.getTypeDataAccess(type);
+        cardinalityOfType = valueState.getPopulatedOrdinals().cardinality();
+        maxOrdinalOfType = valueState.maxOrdinal();
+
+        // initialize the prefix index.
+        build();
+    }
+
+    private void build() {
+
+        // tell memory recycler to use current tst's long arrays next time when long array is requested.
+        // note reuse only happens once swap is called and bits are reset
+        if (prefixIndex != null) prefixIndex.recycleMemory(memoryRecycle);
+
+        Tst tst = new Tst(maxOrdinalOfType, cardinalityOfType, cardinalityOfKeyField, memoryRecycle);
+        BitSet ordinals = readStateEngine.getTypeState(type).getPopulatedOrdinals();
+        int ordinal = ordinals.nextSetBit(0);
+        while (ordinal != -1) {
+            String[] keys = getKey(ordinal);
+            for (String key : keys)
+                tst.insert(key, ordinal);
+            ordinal = ordinals.nextSetBit(ordinal + 1);
+        }
+
+        prefixIndex = tst;
+        prefixIndexVolatile = tst;
+        // safe to return previous long arrays on next request for long array.
+        memoryRecycle.swap();
+    }
+
+    /**
+     * Return the key to index in prefix index. Override this method to support tokens for the key. By default keys are indexed as lower case characters.
```

Comment:
Seems simple enough to access this functionality, and it would be really neat to see it in action and verify that it works.  Does it make sense to add a test for the tokenization, and perhaps a simple tokenization scheme (e.g. split words by whitespace)?